master/custom/factories.py

```diff
@@ -893,7 +893,9 @@ def setup(self, parallel, branch, test_with_PTY=False, **kwargs):
                 warnOnFailure=True,
             )
         )
-        self.host_configure_cmd.append(util.Interpolate("CONFIG_SITE=%(prop:config_site)s"))
+        self.compile_environ.update(
```

Comment:
I think we should still be defensive about mutating the class member.  I think this should do it (untested):
```suggestion
        self.compile_environ = super().compile_environ.copy()
        self.compile_environ.update(
```

Reply:
Possibly just `self.compile_environ.copy()` rather than `super()`.